### ISSUE TITLE
Respecify the location of the PDF

### ISSUE THREAD

**Issue by wwjCMP** [elias-sundqvist/obsidian-annotator#96]
How can I reconnect the PDF file to my notes after it has been moved to a new location?

**Comment by richukuttan** [comment]
There is a really simple workaround I found for pdfs at least (I'm not sure how epubs are rendered, and haven't played with them much). Go to the md file by clicking the three dots at the top right, and clicking "Open as md". To be completely safe, I'd suggest keeping a backup of the existing md before continuing. Next, you can use ctrl+h to open the "Find and Replace" option, enter the old path in the "Find" field (would be present in any existing annotation that is not working) and the new path in the "Replace" field, and replace all.

Fyi, the file path is stored 3 times within each json - in uri, in document.link.href and in target.source.

@elias-sundqvist , I wonder if the large amount of duplication can be eliminated somehow, maybe by specifying the target path in one (auto-filled) YAML or dataview component of the markdown note and using that in the annotator backend. After all, I do not see any logical reason to have annotations from different pdfs in the same markdown note.

**Comment by elias-sundqvist** [comment]
> @elias-sundqvist , I wonder if the large amount of duplication can be eliminated somehow, maybe by specifying the target path in one (auto-filled) YAML or dataview component of the markdown note and using that in the annotator backend. After all, I do not see any logical reason to have annotations from different pdfs in the same markdown note. I realize that this might take away from the hypothes.is notation, but maybe you can add a button to export the data back into the correct format if required - it should just be pretty simple regex expansion, I think. In the meanwhile, it would make the md file much more palatable.

Yes, for pdfs something like this could be possible. For epubs it will be trickier, since different pages have different urls.  I'll think about it.
